### PR TITLE
Added support for iPhone X Safe Area.

### PR DESCRIPTION
**Pull request by FabriBertani**
✨ **What kind of change does this PR introduce?**

This is a feature

💥 **Does this PR introduce a breaking change?**

No

📝 **Links to relevant issues/docs**

[#46](https://github.com/FluentLayout/Cirrious.FluentLayout/issues/46)

**⤵️ What is the current behavior?**

No support for View.SafeAreaLayoutGuide 

🆕 **What is the new behavior?**

Added support to iPhone X and iOS 11 SafeArea on AdvanceFluentLayoutExtesions: AtTopOfWithSafeArea, AtLeftOfWithSafeArea, AtRightOfWithSafeArea, AtBottomOfWithSafeArea. **Also this could be added to the original AtTopOf, AtLeftOf, etc, instead of the new ones.**

### CODE REVIEW

**Comment by FabriBertani** [comment]
@gshackles I've added ViewWithSafeArea sample to quick page, also update QuickLayout MinimumOSVersion from 7.0 to 8.0 and added LaunchScreen storyboard to avoid iPhone X not using entire screen issue.

File: Cirrious.FluentLayout/AdvancedFluentLayoutExtensions.cs [hunk 1]
```diff
@@ -15,21 +15,41 @@ namespace Cirrious.FluentLayouts.Touch
...
+            UIDevice.CurrentDevice.CheckSystemVersion(11, 0)
+                    ? view.Left().EqualTo().LeftOf(parentView.SafeAreaLayoutGuide).Plus(margin.GetValueOrDefault(DefaultMargin))
+                    : view.AtLeftOf(parentView, margin);
+
         public static FluentLayout AtRightOf(this UIView view, UIView parentView, nfloat? margin = null) =>
 			view.Right().EqualTo().RightOf(parentView).Minus(margin.GetValueOrDefault(DefaultMargin));
 
+        public static FluentLayout AtRightOfWithSafeArea(this UIView view, UIView parentView, nfloat? margin = null) =>
+            UIDevice.CurrentDevice.CheckSystemVersion(11, 0)
+                    ? view.Right().EqualTo().RightOf(parentView.SafeAreaLayoutGuide).Plus(margin.GetValueOrDefault(DefaultMargin))
```
Comment (fedemkr) [inline Cirrious.FluentLayout/AdvancedFluentLayoutExtensions.cs]: In order to be used as `AtRightOf` this should use `Minus` for the margin:

```suggestion
                    ? view.Right().EqualTo().RightOf(parentView.SafeAreaLayoutGuide).Minus(margin.GetValueOrDefault(DefaultMargin))
```

File: Cirrious.FluentLayout/AdvancedFluentLayoutExtensions.cs [hunk 2]
```diff
@@ -15,21 +15,41 @@ namespace Cirrious.FluentLayouts.Touch
...
+                    : view.AtRightOf(parentView, margin);
+
         public static FluentLayout AtBottomOf(this UIView view, UIView parentView, nfloat? margin = null) =>
 			view.Bottom().EqualTo().BottomOf(parentView).Minus(margin.GetValueOrDefault(DefaultMargin));
 
+        public static FluentLayout AtBottomOfWithSafeArea(this UIView view, UIView parentView, nfloat? margin = null) =>
+            UIDevice.CurrentDevice.CheckSystemVersion(11, 0)
+                    ? view.Bottom().EqualTo().BottomOf(parentView.SafeAreaLayoutGuide).Plus(margin.GetValueOrDefault(DefaultMargin))
```
Comment (fedemkr) [inline Cirrious.FluentLayout/AdvancedFluentLayoutExtensions.cs]: In order to be used as `AtBottomOf` this should use `Minus` for the margin:

```suggestion
                    ? view.Bottom().EqualTo().BottomOf(parentView.SafeAreaLayoutGuide).Minus(margin.GetValueOrDefault(DefaultMargin))
```

**Comment by gshackles** [comment]
Sorry for the delay, and thanks for contributing this? Going to tweak the method names slightly, but I'll do that separately and pull this in now :shipit: 🚀

**Pull request merged**